sql/analyzer/optimize_joins.go

```diff
@@ -46,16 +46,11 @@ func optimizeJoins(ctx *sql.Context, a *Analyzer, n sql.Node) (sql.Node, error)
 		return nil, err
 	}
 
-	a.Log("replacing InnerJoins with IndexJoins")
-	return transformInnerJoins(a, n, indexes, aliases)
+	a.Log("replacing LeftJoin,RightJoin,InnerJoin with IndexJoin")
+	return transformJoins(a, n, indexes, aliases)
 }
 
-func transformInnerJoins(
-		a *Analyzer,
-		n sql.Node,
-		indexes []sql.Index,
-		aliases Aliases,
-) (sql.Node, error) {
+func transformJoins(a *Analyzer, n sql.Node, indexes map[string]sql.Index, aliases Aliases, ) (sql.Node, error) {
```

Comment:
nit: `aliases Aliases, )` more idiomatic if this isn't multi-line...

Reply:
fixed